CMakeLists.txt

```diff
@@ -4,13 +4,74 @@ project(mrover)
 ## Compile as C++17, supported in ROS Kinetic and newer
 add_compile_options(-std=c++17 -Wall -Wextra -Werror -pedantic)
 
-## Find catkin macros and libraries
-## if COMPONENTS list like find_package(catkin REQUIRED COMPONENTS xyz)
-## is used, also find other catkin packages
-find_package(catkin REQUIRED COMPONENTS
+# Generate lists of universal packages to find
+
+# find_package
+set(_PACKAGES
+    rospy
+    std_msgs
+    message_generation
+    dynamic_reconfigure
+    rostest
+    )
+
+# link_directories
+set(_LINK_DIRS "")
+
+# message files
+set(_MESSAGE_FILES 
+    Course.msg
+    Waypoint.msg
+    Chassis.msg
+    )
+
+# service files
+set(_SERVICE_FILES
+    PublishCourse.srv
+    )
+
+# generate messages list
+set(_MESSAGES
+    std_msgs
+    )
+
+# dynamic reconfigure
+set(_DYNAMIC_RECON
+    config/DetectorParams.cfg
+    )
+
+# catkin_package
+set(_CAT_PACKAGES
+    roscpp rospy std_msgs message_runtime
+    )
+
+# launch install macro
+macro(install_launch_macro)
+    install(DIRECTORY launch/
+        DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/launch
+        )
+endmacro()
+
+# Subdirectories before message declarations
+set(_SUBDIR "")
+
+
+
+# -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+# Specify build details by appending to lists
+# and implementing the some extra macros
+# Add new devices as elseif blocks
+# Select device with --cmake-flags -D DEVICE=<fill in device>
+# -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-
+
+if(${DEVICE} MATCHES "raspi4")
+    # Add any raspi4 specific packages here
+    # list(APPEND _PACKAGES )
+
+else()
```

Comment:
Eventually should be else if but that can go in a different pr if you don't know what to change

Reply:
Actually maybe not. Depends on if we want different stuff for base station vs Jetson. Default should be base station if we do make it diff